diff --git a/dom-monkeypatch.d.ts b/dom-monkeypatch.d.ts
index 20cbd6a7723d42d5f68ee93ad4ab1c9557d42a88..abeb2813f479a16059297a53ab8aae6c664c9d80 100755
--- a/dom-monkeypatch.d.ts
+++ b/dom-monkeypatch.d.ts
@@ -39,10 +39,10 @@ interface Event {
    * "submit".
    */
   readonly type: string;
-  readonly AT_TARGET: 2;
-  readonly BUBBLING_PHASE: 3;
-  readonly CAPTURING_PHASE: 1;
-  readonly NONE: 0;
+  readonly AT_TARGET: number;
+  readonly BUBBLING_PHASE: number;
+  readonly CAPTURING_PHASE: number;
+  readonly NONE: number;
   composedPath(): any[];
   initEvent(type: string, bubbles?: boolean, cancelable?: boolean): void;
   preventDefault(): void;