diff --git a/dist/ember-template-compiler.js b/dist/ember-template-compiler.js
index 485c2db34f559e1c42ba1b515e8ac6959e2de783..245ebe458fbe69cf494c7ec5b03570b4a337ad2a 100644
--- a/dist/ember-template-compiler.js
+++ b/dist/ember-template-compiler.js
@@ -6942,7 +6942,7 @@ define("@glimmer/syntax", ["exports", "@glimmer/util", "@handlebars/parser", "si
       if (true /* DEBUG */) {
         if (expected !== undefined && chars !== expected) {
           // eslint-disable-next-line no-console
-          console.warn("unexpectedly found " + JSON.stringify(chars) + " when slicing source, but expected " + JSON.stringify(expected));
+          // console.warn("unexpectedly found " + JSON.stringify(chars) + " when slicing source, but expected " + JSON.stringify(expected));
         }
       }
       return new SourceSlice({
diff --git a/dist/packages/@ember/-internals/glimmer/index.js b/dist/packages/@ember/-internals/glimmer/index.js
index 110720eee3522722af6e77732ef1ae10292b06af..514254c6a20af851d8e5b6fb6a55e259183fc3cd 100644
--- a/dist/packages/@ember/-internals/glimmer/index.js
+++ b/dist/packages/@ember/-internals/glimmer/index.js
@@ -4955,6 +4955,16 @@ function setupApplicationRegistry(registry) {
       let owner = getOwner(props);
       assert('DomBuilderService is unexpectedly missing an owner', owner);
       let env = owner.lookup('-environment:main');
+
+      console.error('checking for render mode');
+      if (!env._renderMode && typeof globalThis !== 'undefined') {
+        env._renderMode = globalThis.__boxelRenderMode;
+        console.error('render mode changed');
+      }
+
+      console.error('render mode is:');
+      console.error(env._renderMode);
+
       switch (env._renderMode) {
         case 'serialize':
           return serializeBuilder.bind(null);
