diff --git a/src/preprocessor/noop.js b/src/preprocessor/noop.js
--- a/src/preprocessor/noop.js
+++ b/src/preprocessor/noop.js
@@ -25,6 +25,8 @@ export const postprocess = (messages, fileName) => {
     msgs[0].message +=
       'To lint Gjs/Gts files please follow the setup guide at https://github.com/ember-cli/eslint-plugin-ember#gtsgjs' +
       '\nNote that this error can also happen if you have multiple versions of eslint-plugin-ember in your node_modules';
+    // We want this to fail in CI so we notice if linting is broken
+    msgs[0].fatal = true;
   }
   parsedFiles.delete(fileName); // required for tests
   return msgs;
