diff --git a/dist/ember-template-compiler.js b/dist/ember-template-compiler.js
index 485c2db34f559e1c42ba1b515e8ac6959e2de783..245ebe458fbe69cf494c7ec5b03570b4a337ad2a 100644
--- a/dist/ember-template-compiler.js
+++ b/dist/ember-template-compiler.js
@@ -6942,7 +6942,7 @@ define("@glimmer/syntax", ["exports", "@glimmer/util", "@handlebars/parser", "si
       if (true /* DEBUG */) {
         if (expected !== undefined && chars !== expected) {
           // eslint-disable-next-line no-console
-          console.warn("unexpectedly found " + JSON.stringify(chars) + " when slicing source, but expected " + JSON.stringify(expected));
+          // console.warn("unexpectedly found " + JSON.stringify(chars) + " when slicing source, but expected " + JSON.stringify(expected));
         }
       }
       return new SourceSlice({
