diff --git a/dist/index.js b/dist/index.js
index 22f971126bf9375e1fb7a85c3627ab69050467e1..95bb4004a6941f37982a01e46522d71a8b1f0dd9 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -215,4 +215,5 @@ exports.buildASTPlugin = buildASTPlugin;
 exports.decodeScopedCSSRequest = decodeScopedCSSRequest;
 exports.installScopedCSS = installScopedCSS;
 exports.isScopedCSSRequest = isScopedCSSRequest;
+exports.scopedCSSTransform = scopedCSSTransform;
 //# sourceMappingURL=index.js.map