diff --git a/dist/webpack.js b/dist/webpack.js
index 3bc3fac9597f065c895c633d5ccd4d851657049a..1103bcd79654ac641a3b1637f234ba6cf1365b6c 100644
--- a/dist/webpack.js
+++ b/dist/webpack.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, '__esModule', { value: true });
 var path = require('path');
 var index = require('./index.js');
 
-const virtualLoaderName = 'glimmer-scoped-css';
+const virtualLoaderName = 'glimmer-scoped-css/virtual-loader';
 class GlimmerScopedCSSWebpackPlugin {
   addLoaderAlias(compiler, name, alias) {
     let {