diff --git a/lib/core/index.js b/lib/core/index.js
index 920e55b9c01222c280b7f629ab4c8aaac8a987f7..a2464e449d59cf184ff89ac53f048639b541310e 100644
--- a/lib/core/index.js
+++ b/lib/core/index.js
@@ -319,6 +319,8 @@ module.exports = function createMiddleware(_dir, _options) {
       res.setHeader('content-length', stat.size);
       res.setHeader('content-type', contentType);
 
+      res.setHeader('Origin-Agent-Cluster', '?0');
+
       // set the response statusCode if we have a request statusCode.
       // This only can happen if we have a 404 with some kind of 404.html
       // In all other cases where we have a file we serve the 200