diff --git a/lib/crypto/index.js b/lib/crypto/index.js
index d1579c21535a3c3bed0b42fb419c4e17e2fc8050..f7a6ba90b899262444d20be636a443697d1ab11b 100644
--- a/lib/crypto/index.js
+++ b/lib/crypto/index.js
@@ -1707,10 +1707,14 @@ class Crypto extends _typedEventEmitter.TypedEventEmitter {
     }
     const res = await this.baseApis.uploadKeysRequest(requestBody);
     if (fallbackJson) {
-      this.fallbackCleanup = setTimeout(() => {
+      let timer = setTimeout(() => {
         delete this.fallbackCleanup;
         this.olmDevice.forgetOldFallbackKey();
       }, 60 * 60 * 1000);
+      if (typeof timer.unref === 'function') {
+        timer.unref();
+      }
+      this.fallbackCleanup = timer;
     }
     await this.olmDevice.markKeysAsPublished();
     return res;
diff --git a/lib/http-api/utils.js b/lib/http-api/utils.js
index 9115598fd769bad15ed23293a68d617d5dbf21a5..95c28e59d8414dbc0837f5c85c98dd96a80446c8 100644
--- a/lib/http-api/utils.js
+++ b/lib/http-api/utils.js
@@ -30,9 +30,12 @@ limitations under the License.
 // Ponyfill for https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/timeout
 function timeoutSignal(ms) {
   const controller = new AbortController();
-  setTimeout(() => {
+  let timer = setTimeout(() => {
     controller.abort();
   }, ms);
+  if (typeof timer.unref === 'function') {
+    timer.unref();
+  }
   return controller.signal;
 }
 function anySignal(signals) {